### PR TITLE
main: show compatible in output of rauc status

### PR DESCRIPTION
**Pull request by ejoerns**
The compatible of the device rauc is running on is an information of
certain interest for both user of rauc and possible scripts.

Signed-off-by: Enrico Jorns <ejo@pengutronix.de>

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -551,6 +552,7 @@ static gchar* r_status_formatter_shell(void)
 	gint slotcnt = 0;
 	GString *text = g_string_new(NULL);
 
+	g_string_append_printf(text, "RAUC_SYSTEM_COMPATIBLE=%s\n", r_context()->config->system_compatible);
```
Comment (jluebbe) [inline src/main.c]: The system compatible may contains spaces (or other characters), so this needs to be shell-escaped.

File: src/main.c [hunk 2]
```diff
@@ -267,25 +267,67 @@ static gboolean checksum_start(int argc, char **argv)
...
 	GString *text = g_string_new(NULL);
+	GPtrArray *hooks = NULL;
 	gint cnt;
 
-	g_string_append_printf(text, "RAUC_MF_COMPATIBLE=%s\n", manifest->update_compatible);
-	g_string_append_printf(text, "RAUC_MF_VERSION=%s\n", manifest->update_version ?: "");
-	g_string_append_printf(text, "RAUC_MF_DESCRIPTION=%s\n", manifest->update_description ?: "");
-	g_string_append_printf(text, "RAUC_MF_BUILD=%s\n", manifest->update_build ?: "");
+	formatter_shell_append(text, "RAUC_MF_COMPATIBLE", manifest->update_compatible);
+	formatter_shell_append(text, "RAUC_MF_VERSION", manifest->update_version ?: "");
```
Comment (jluebbe) [inline src/main.c]: NULL values are already handled by the helper function.

File: src/main.c [hunk 3]
```diff
@@ -267,25 +267,67 @@ static gboolean checksum_start(int argc, char **argv)
...
 	g_string_append_printf(text, "RAUC_MF_IMAGES=%d\n", g_list_length(manifest->images));
 	g_string_append_printf(text, "RAUC_MF_FILES=%d\n", g_list_length(manifest->files));
 
+	hooks = g_ptr_array_new();
+	if (manifest->hooks.install_check == TRUE) {
+		g_ptr_array_add(hooks, g_strdup("install-check"));
+	}
+	g_ptr_array_add(hooks, NULL);
+
+	formatter_shell_append(text, "RAUC_MF_HOOKS", g_strjoinv(" ", (gchar**) hooks->pdata));
```
Comment (jluebbe) [inline src/main.c]: The return value of g_strjoinv is leaked here.

File: src/main.c [hunk 4]
```diff
@@ -306,22 +348,51 @@ static gchar *info_formatter_shell(RaucManifest *manifest)
...
+	g_string_append_printf(text, "Hooks:      \t'%s'\n", g_strjoinv(" ", (gchar**) hooks->pdata));
+
+	g_ptr_array_unref(hooks);
+
 	cnt = g_list_length(manifest->images);
 	g_string_append_printf(text, "%d Image%s%s\n", cnt, cnt == 1 ? "" : "s", cnt > 0 ? ":" : "");
 	cnt = 1;
 	for (GList *l = manifest->images; l != NULL; l = l->next) {
 		RaucImage *img = l->data;
+		//GPtrArray *hooks = NULL;
```
Comment (jluebbe) [inline src/main.c]: Drop this comment.

**Comment by ejoerns** [comment]
Ping :)

**Pull request merged**